### ISSUE TITLE
why not using one-hot encoding and softmax for CIFAR-10 model

### ISSUE THREAD

**Issue by CunliangGeng** [carpentries-incubator/deep-learning-intro#292]
In the [material](https://carpentries-incubator.github.io/deep-learning-intro/04-advanced-layer-types/index.html), it's not mentioned why one-hot encoding is not used and why output layer has 10 units but not 1.

The material has 
> To calculate crossentropy loss for data that has its classes represented by integers (i.e., not one-hot encoded), we use the SparseCategoricalCrossentropy() function:

but it's too late to introduce here. 

We should explain why one-hot encoding is not used at the point where building the architecture.

Reference:
1. https://keras.io/api/losses/probabilistic_losses/#sparsecategoricalcrossentropy-class

**Comment by svenvanderburg** [comment]
I agree we should explain this, and link to one-hot encoding.
